packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/ShadowNodeWrapper.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ *  * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ *  * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ *  * Copyright 2016-Present Datadog, Inc.
+ *
+ */
+
+package com.datadog.reactnative.sessionreplay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.reactnative.sessionreplay.utils.ReflectionUtils
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.ReactShadowNode
+import com.facebook.react.uimanager.UIImplementation
+import com.facebook.react.uimanager.UIManagerModule
+import java.util.concurrent.CountDownLatch
+
+internal class ShadowNodeWrapper(
+    private val logger: InternalLogger,
+    private val reactContext: ReactContext,
+    private val uiManagerModule: UIManagerModule?,
+    private val reflectionUtils: ReflectionUtils = ReflectionUtils(logger)
+) {
+    private var shadowNode: ReactShadowNode<out ReactShadowNode<*>>? = null
+
+    internal fun initialize(viewId: Int): Boolean {
+        shadowNode = getShadowNode(viewId)
+        return shadowNode != null
+    }
+
+    internal fun getDeclaredShadowNodeField(fieldName: String): Any? {
+        return reflectionUtils.getDeclaredField(
+            shadowNode?.javaClass?.superclass as Class<*>,
+            shadowNode,
+            fieldName
+        )
+    }
+
+    private fun getShadowNode(viewId: Int): ReactShadowNode<out ReactShadowNode<*>>? {
```

Comment:
I would propose to make this method `static` and `internal` returning `ShadowNodeWrapper`, and then follow comment in `ReactTextPropertiesResolver`